toolkit/tools/pkg/imagecustomizerlib/extractpartitions.go

```diff
@@ -79,11 +121,72 @@ func compressWithZstd(partitionRawFilepath string) (partitionFilepath string, er
 		return "", fmt.Errorf("failed to compress %s with zstd:\n%w", partitionRawFilepath, err)
 	}
 
-	// Remove raw file since output partition format is raw-zst.
-	err = os.Remove(partitionRawFilepath)
+	return partitionRawFilepath + ".zst", nil
+}
+
+// Prepend a skippable frame with the metadata to the specified partition file.
+func addSkippableFrame(tempPartitionFilepath string, skippableFrameMetadata [SkippableFramePayloadSize]byte, partitionFilename string, outDir string) (partitionFilepath string, err error) {
+	// Open tempPartitionFile for reading
+	tempPartitionFile, err := os.OpenFile(tempPartitionFilepath, os.O_RDWR, PartitionFilePermissions)
 	if err != nil {
-		return "", fmt.Errorf("failed to remove raw file %s:\n%w", partitionRawFilepath, err)
+		return "", fmt.Errorf("failed to open partition file %s:\n%w", tempPartitionFilepath, err)
 	}
+	// Create a skippable frame
+	skippableFrame := createSkippableFrame(SkippableFrameMagicNumber, SkippableFramePayloadSize, skippableFrameMetadata)
+	// Define the final partition file path
+	partitionFilepath = outDir + "/" + partitionFilename + "_final" + ".raw.zst"
+	// Create partition file
```

Comment:
I would recommend writing a Prepend() function to handle all of this such that the resulting file name is the same
something like:
1) create a new temp file
2) add the data you want to prepend to it
3) add the data from old file to new file
4) delete the old file
5) rename / mv the temp file to the original filename
you could add this to file.go perhaps
https://github.com/microsoft/azurelinux/blob/4cb4cf97332d38647b4648157e0a826cc6ff73e9/toolkit/tools/internal/file/file.go